### ISSUE TITLE
Weird error when using enum(val1, val2)

### ISSUE THREAD

**Issue by wimglenn** [python/cpython#107878]
Using the example from the [docs](https://docs.python.org/3/library/enum.html):

```
class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3
```

Actual output (py 3.11):

```
>>> Color(0, 1)
TypeError: <enum 0> cannot extend <enum 'Color'>
```

Expected output- not 100% sure but, something along the lines of:
- `TypeError` saying "... _takes 1 positional argument but 2 were given_" like `Color()` does, or
- `ValueError` saying "... _is not a valid Color_" like `Color(0)` does

The current metaclass error message is a bit confusing.

**Comment by JimZhouZZY** [comment]
The error message is correct.
`Color(0, 1)` calls for creating a new enum subclass with `value` `0` and `names` `1`. See enum.py line700
```python
    def __call__(cls, value, names=None, *, module=None, qualname=None, type=None, start=1):
        """
        Either returns an existing member, or creates a new enum class.

        This method is used both when an enum class is given a value to match
        to an enumeration member (i.e. Color(3)) and for the functional API
        (i.e. Color = Enum('Color', names='RED GREEN BLUE')).

        When used for the functional API:

        `value` will be the name of the new class.

        `names` should be either a string of white-space/comma delimited names
        (values will start at `start`), or an iterator/mapping of name, value pairs.

        ...
        """
        if names is None:  # simple value lookup
            return cls.__new__(cls, value)
        # otherwise, functional API: we're creating a new Enum type
        return cls._create_(value, names, ...)
```
According to the [doc](https://docs.python.org/3/howto/enum.html#restricted-enum-subclassing), subclassing an enumeration is allowed only if the enumeration does not define any members. So in enum.py line930 it raised the error.

**Comment by wimglenn** [comment]
I see that, but it's a leaky abstraction.

**Comment by ethanfurman** [comment]
This is fixed in 3.13:

    ValueError: (0, 1) is not a valid Color

**Comment by wimglenn** [comment]
Right, apologies for not checking main before reporting. Looks fixed in 3.12.0rc1 too. Git bisected it to https://github.com/python/cpython/commit/65dab1506e9c8ac40b3f9da834121d86c6417275. Closing as duplicate of https://github.com/python/cpython/issues/92647
